sim/battle.ts

```diff
@@ -2533,6 +2533,7 @@ export class Battle {
 
 	runAction(action: Action) {
 		const pokemonOriginalHP = action.pokemon?.hp;
+		let residualPokemon: (readonly [Pokemon, number])[] | null = null;
```

Comment:
Could default this to `[]` perhaps?

Reply:
I don’t see the advantage of that. You’d just be checking `residualPokemon.length` instead of `residualPokemon`.

Reply:
Except that the `for` loop does that _for_ you anyway...

Reply:
Oh, right, I always forget for ofs work like that. That seems neater, then.